src/ir_MilesTag2.cpp

```diff
@@ -0,0 +1,155 @@
+// Copyright 2020 Victor Mukayev (vitos1k)
+/// @file
+/// @brief Support for the MilesTag2 IR protocol for LaserTag gaming
+/// @see http://hosting.cmalton.me.uk/chrism/lasertag/MT2Proto.pdf
+
+// Supports:
+//   Brand: Theoretically,  Model: MILESTAG2 SUPPORTED HARDWARE
+
+// TODO(vitos1k): This implementation would support only
+// short SHOT packets(14bits) and MSGs = 24bits. Support
+// for long MSGs > 24bits is TODO
+
+#include <algorithm>
+#include "IRrecv.h"
+#include "IRsend.h"
+#include "IRutils.h"
+
+// Constants
+// Constants are similar to Sony SIRC protocol, bassically they are very
+// similar, just nbits are varying
+
+const uint16_t kMilesHdrMark = 2400;
+const uint16_t kMilesSpace = 600;
+const uint16_t kMilesOneMark = 1200;
+const uint16_t kMilesZeroMark = 600;
+const uint16_t kMilesRptLength = 32000;
+const uint16_t kMilesMinGap = 32000;
+const uint16_t kMilesStdFreq = 38000;  // kHz
+const uint16_t kMilesStdDuty = 25;
+
+
+
+#if SEND_MILESTAG2
+/// Send a MilesTag2 formatted Shot packet.
+/// Status: NEEDS TESTING
+/// @param[in] data The message to be sent.
+/// @param[in] nbits The number of bits of message to be sent.
+/// @param[in] repeat The number of times the command is to be repeated.
+void IRsend::sendMilesShot(const uint64_t data, const uint16_t nbits,
+                      const uint16_t repeat) {
+    _sendMiles(data, nbits, repeat);
+}
+
+/// Send a MilesTag2 formatted MESSAGE packet.
+/// Status: NEEDS TESTING
+/// @param[in] data The message to be sent.
+/// @param[in] nbits The number of bits of message to be sent.
+/// @param[in] repeat The number of times the command is to be repeated.
+void IRsend::sendMilesMsg(const uint64_t data, const uint16_t nbits,
+                      const uint16_t repeat) {
+    _sendMiles(data, nbits, repeat);
+}
+
+void IRsend::_sendMiles(const uint64_t data, const uint16_t nbits,
+                      const uint16_t repeat) {
```

Comment:
Let's ditch the separate `send()` routines for msg & shot. Use a common `sendMilesTag2()` instead and let the user of that function choose what one they want by specifying the correct bit size.